tests/pages/page.py

```diff
@@ -0,0 +1,42 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+TIMEOUT = 10
+
+
+class Page(object):
+
+    _url = None
+
+    def __init__(self, base_url, selenium):
+        self.base_url = base_url
+        self.selenium = selenium
+        self.timeout = TIMEOUT
+
+    def open(self):
+        self.selenium.get(self.url)
+        self.wait_for_page_to_load()
+        return self
+
+    @property
+    def url(self):
+        if self._url is not None:
+            return self._url.format(base_url=self.base_url)
+        return self.base_url
+
+    def wait_for_page_to_load(self):
+        pass
+
+
+class PageRegion(object):
+
+    _root_locator = None
+
+    def __init__(self, selenium, root=None):
+        self.selenium = selenium
+        self.timeout = TIMEOUT
+        self.root = root
+
+        if self.root is None and self._root_locator is not None:
+            self.root = self.selenium.find_element(*self._root_locator)
```

Comment:
I'm not keen on `self.root` defaulting to `self.selenium`, though perhaps we could default it to the document root (if that's even possible?). I was thinking that root is totally optional, and for page regions that don't specify it, using `region.root` would not make sense. Instead those page regions should use `self.selenium` directly.